stage.py

```diff
@@ -39,13 +50,65 @@ def verbose_str(self):
         max_index = i
     return "%s\n    Longest Task: %s" % (self, self.tasks[i])
 
-  def load_balancing_badness(self):
+  def get_executor_id_to_resource_metrics(self):
+    """
+    Returns a mapping from executor id to a description of how each of its resources was used while
+    this stage's tasks were running on that executor.
+    """
+    return {executor: Stage.__get_resource_metrics_for_executor_tasks(tasks)
+      for executor, tasks in self.get_executor_id_to_tasks().iteritems()}
+
+  @staticmethod
+  def __get_resource_metrics_for_executor_tasks(tasks):
```

Comment:
Can you move this whole function to the ResourceMetrics class, so the code about resource metrics is mostly contained within that class?